src/Storage/Repository/PgDataRepository.cs

```diff
@@ -180,6 +180,34 @@ await using (NpgsqlDataReader reader = await pgcom.ExecuteReaderAsync())
 
         /// <inheritdoc/>
         public async Task<DataElement> Update(Guid instanceGuid, Guid dataElementId, Dictionary<string, object> propertylist)
+        {
+            // Retry loop below because Postgres doesn't lock the element between the read and the following update
+            // even if it's in a transaction with repeatable read.
+            PostgresException exceptionToRethrow = null;
+            for (int i = 0; i < 10; i++)
+            {
+                try
+                {
+                    return await UpdateInternal(instanceGuid, dataElementId, propertylist);
+                }
+                catch (PostgresException e)
+                {
+                    if (e.Message == "40001: could not serialize access due to concurrent update")
+                    {
+                        exceptionToRethrow = e;
+                        await Task.Delay(100);
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+            }
+
+            throw exceptionToRethrow;
+        }
+
+        private async Task<DataElement> UpdateInternal(Guid instanceGuid, Guid dataElementId, Dictionary<string, object> propertylist)
```

Comment:
The `instanceGuid` parameter is unnecessary.

Reply:
Done